From: Chris Lalancette <clalance@redhat.com>
Date: Fri, 6 Mar 2009 09:04:59 +0100
Subject: [xen] silence MMCONFIG warnings
Message-id: 49B0D92B.6010003@redhat.com
O-Subject: [RHEL5.4 PATCH v2]: Silence MMCONFIG warnings under Xen
Bugzilla: 462572
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Justin M. Forbes <jforbes@redhat.com>

All,
     When you boot up the Xen kernel as either a dom0 or a domU, the boot
messages always include:

PCI: BIOS Bug: MCFG area at e0000000 is not E820-reserved

That's because the e820 mapped passed from the hypervisor to any domain,
including dom0, is always one large RAM region; there are no holes.  Silence
this totally bogus warning on Xen, since it is not a bug there. (v2: only
silence the "BIOS BUG" message, but still print the "Not using MMCONFIG" for
debugging)

This should address BZ 462572.  Please review and ACK.

--
Chris Lalancette

diff --git a/arch/i386/pci/mmconfig.c b/arch/i386/pci/mmconfig.c
index 1d4a73d..c659d99 100644
--- a/arch/i386/pci/mmconfig.c
+++ b/arch/i386/pci/mmconfig.c
@@ -163,8 +163,10 @@ void __init pci_mmcfg_init(void)
 	if (!e820_all_mapped(pci_mmcfg_config[0].base_address,
 			pci_mmcfg_config[0].base_address + MMCONFIG_APER_MIN,
 			E820_RESERVED)) {
+#ifndef CONFIG_XEN
 		printk(KERN_ERR "PCI: BIOS Bug: MCFG area at %x is not E820-reserved\n",
 				pci_mmcfg_config[0].base_address);
+#endif
 		printk(KERN_ERR "PCI: Not using MMCONFIG.\n");
 		return;
 	}
diff --git a/arch/x86_64/pci/mmconfig.c b/arch/x86_64/pci/mmconfig.c
index ec40acb..b36374d 100644
--- a/arch/x86_64/pci/mmconfig.c
+++ b/arch/x86_64/pci/mmconfig.c
@@ -150,8 +150,10 @@ void __init pci_mmcfg_init(void)
 	if (!e820_all_mapped(pci_mmcfg_config[0].base_address,
 			pci_mmcfg_config[0].base_address + MMCONFIG_APER_MIN,
 			E820_RESERVED)) {
+#ifndef CONFIG_XEN
 		printk(KERN_ERR "PCI: BIOS Bug: MCFG area at %x is not E820-reserved\n",
 				pci_mmcfg_config[0].base_address);
+#endif
 		printk(KERN_ERR "PCI: Not using MMCONFIG.\n");
 		return;
 	}
